cmd/fuego/templates/controller/service.go

```diff
@@ -0,0 +1,90 @@
+package controller
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/go-fuego/fuego"
+)
+
+type NewControllerServiceImpl struct {
+	newControllerRepository map[string]NewController
+	mu                      sync.RWMutex
```

Comment:
Nice, you could have used [`sync.Map`](https://pkg.go.dev/sync#Map) to not be bothered with Mutexes, but it works!

Reply:
`sync.Map` does not work with generics as far as I know. I didn't want to add noisy code to wrap `sync.Map` around a generic compliant interface.